platform/api.progress/src/org/netbeans/modules/progress/spi/Controller.java

```diff
@@ -344,6 +344,10 @@ else if (event.getType() == ProgressEvent.TYPE_FINISH &&
                         if (lastEvent.isSwitched()) {
                             event.markAsSwitched();
                         }
+                        // preserve finish type
+                        if (lastEvent.getType() == ProgressEvent.TYPE_FINISH) {
```

Comment:
Great, finish event ***must*** be delivered, otherwise many things in UI will break. Though some tests could be useful to verify correct merge effects on the various other types.

Reply:
Since the module contains no tests for this functionality, there is nothing that can be easily augmented for this particular change.